File: Juego.py
# ==============================================================
# RPG SOCIAL MANAGER - VERSION TKINTER MEJORADA
# Con efectos especiales:
# ✔ Flash al atacar
# ✔ Sacudida enemigo
# ✔ Curación verde
# ✔ Barras animadas
# ✔ Interfaz gamer neon
# ✔ Log estilo consola
# ==============================================================

import tkinter as tk
from tkinter import ttk, messagebox
import random

# ==============================================================
# CLASE PERSONAJE
# ==============================================================

class Personaje:
    def __init__(self, nombre, hp, dano):
        self.nombre = nombre
        self.hp = hp
        self.hp_max = hp
        self.dano = dano

    def vivo(self):
        return self.hp > 0

    def recibir_dano(self, dmg):
        self.hp = max(0, self.hp - dmg)


# ==============================================================
# JUEGO
# ==============================================================

class Juego:

    def __init__(self, root):

        self.root = root
        self.root.title("⚔ RPG SOCIAL MANAGER ⚔")
        self.root.geometry("1250x760")
        self.root.config(bg="#050505")

        # -----------------------------------
        # PERSONAJES
        # -----------------------------------
        self.est = Personaje("👨‍🎓 Estudiantes", 160, 24)
        self.obr = Personaje("👷 Obreros", 190, 28)
        self.bur = Personaje("🎩 Burguesía", 145, 20)

        self.heroes = [self.est, self.obr, self.bur]

        self.enemy = Personaje("👑 El Caudillo", 500, 35)

        self.crear_ui()
        self.actualizar()

    # ==========================================================
    # UI
    # ==========================================================

    def crear_ui(self):

        titulo = tk.Label(
            self.root,
            text="⚔ RPG SOCIAL MANAGER ⚔",
            font=("Arial", 28),
            fg="#00ffff",
            bg="#2B00E9"
        )
        titulo.pack(pady=10)

        # PANEL PRINCIPAL
        self.main = tk.Frame(self.root, bg="#050505")
        self.main.pack()

        # ------------------------------------------------------
        # HEROES
        # ------------------------------------------------------
        self.panel_heroes = tk.Frame(self.main, bg="#050505")
        self.panel_heroes.grid(row=0, column=0, padx=25)

        self.barras = {}

        for h in self.heroes:

            card = tk.Frame(
                self.panel_heroes,
                bg="#111",
                bd=3,
                relief="ridge",
                padx=15,
                pady=15
            )
            card.pack(pady=15)

            tk.Label(
                card,
                text=h.nombre,
                fg="white",
                bg="#111",
                font=("Arial", 16, "bold")
            ).pack()

            barra = ttk.Progressbar(
                card,
                length=280,
                maximum=h.hp_max
            )
            barra.pack(pady=10)

            txt = tk.Label(
                card,
                text="",
                fg="#00ff00",
                bg="#111",
                font=("Arial", 11)
            )
            txt.pack()

            self.barras[h.nombre] = (barra, txt)

        # ------------------------------------------------------
        # ENEMIGO
        # ------------------------------------------------------
        self.enemy_frame = tk.Frame(
            self.main,
            bg="#220000",
            bd=5,
            relief="ridge",
            padx=40,
            pady=40
        )
        self.enemy_frame.grid(row=0, column=1, padx=50)

        self.enemy_label = tk.Label(
            self.enemy_frame,
            text="👑 EL CAUDILLO 👑",
            fg="red",
            bg="#220000",
            font=("Arial Black", 24)
        )
        self.enemy_label.pack()

        self.enemy_bar = ttk.Progressbar(
            self.enemy_frame,
            length=360,
            maximum=self.enemy.hp_max
        )
        self.enemy_bar.pack(pady=15)

        self.enemy_txt = tk.Label(
            self.enemy_frame,
            text="",
            fg="white",
            bg="#220000",
            font=("Arial", 12)
        )
        self.enemy_txt.pack()

        # ------------------------------------------------------
        # BOTONES
        # ------------------------------------------------------
        btn_frame = tk.Frame(self.root, bg="#050505")
        btn_frame.pack(pady=15)

        self.boton(
            btn_frame, "⚔ Ataque", self.atacar, 0
        )

        self.boton(
            btn_frame, "✨ Especial", self.especial, 1
        )

        self.boton(
            btn_frame, "💚 Curar", self.curar, 2
        )

        self.boton(
            btn_frame, "🔄 Reiniciar", self.reiniciar, 3
        )

        # ------------------------------------------------------
        # LOG
        # ------------------------------------------------------
        self.log = tk.Text(
            self.root,
            height=14,
            bg="black",
            fg="#00ff00",
            font=("Consolas", 11)
        )
        self.log.pack(fill="both", padx=20, pady=12)

    # ==========================================================
    # BOTON BONITO
    # ==========================================================

    def boton(self, frame, texto, comando, col):

        btn = tk.Button(
            frame,
            text=texto,
            width=16,
            height=2,
            bg="#111",
            fg="white",
            font=("Arial", 11, "bold"),
            command=comando,
            activebackground="#00ffff"
        )

        btn.grid(row=0, column=col, padx=8)

        btn.bind("<Enter>", lambda e: btn.config(bg="#00aaaa"))
        btn.bind("<Leave>", lambda e: btn.config(bg="#111"))

    # ==========================================================
    # LOG
    # ==========================================================

    def escribir(self, txt):
        self.log.insert(tk.END, txt + "\n")
        self.log.see(tk.END)

    # ==========================================================
    # ACTUALIZAR
    # ==========================================================

    def actualizar(self):

        for h in self.heroes:
            barra, txt = self.barras[h.nombre]
            barra["value"] = h.hp
            txt.config(text=f"{h.hp}/{h.hp_max}")

        self.enemy_bar["value"] = self.enemy.hp
        self.enemy_txt.config(
            text=f"{self.enemy.hp}/{self.enemy.hp_max}"
        )

    # ==========================================================
    # FLASH COLOR
    # ==========================================================

    def flash_enemy(self, color):

        original = self.enemy_frame["bg"]

        self.enemy_frame.config(bg=color)
        self.enemy_label.config(bg=color)
        self.enemy_txt.config(bg=color)

        self.root.after(
            180,
            lambda: self.enemy_frame.config(bg=original)
        )

        self.root.after(
            180,
            lambda: self.enemy_label.config(bg=original)
        )

        self.root.after(
            180,
            lambda: self.enemy_txt.config(bg=original)
        )

    # ==========================================================
    # SACUDIDA
    # ==========================================================

    def shake_enemy(self):

        x = self.enemy_frame.winfo_x()

        for i in range(6):
            move = -10 if i % 2 == 0 else 10
            self.enemy_frame.place(x=x + move, y=0)
            self.root.update()

        self.enemy_frame.place_forget()
        self.enemy_frame.grid(row=0, column=1, padx=50)

    # ==========================================================
    # ATAQUE
    # ==========================================================

    def atacar(self):

        for h in self.heroes:
            if h.vivo():

                dmg = random.randint(h.dano - 5, h.dano + 5)

                self.enemy.recibir_dano(dmg)

                self.escribir(
                    f"{h.nombre} golpea por {dmg}"
                )

        self.flash_enemy("red")
        self.shake_enemy()

        self.turno_enemy()
        self.verificar()
        self.actualizar()

    # ==========================================================
    # ESPECIAL
    # ==========================================================

    def especial(self):

        total = random.randint(90, 130)

        self.enemy.recibir_dano(total)

        self.escribir(
            f"✨ ATAQUE COMBINADO causa {total} daño"
        )

        self.flash_enemy("purple")
        self.shake_enemy()

        self.turno_enemy()
        self.verificar()
        self.actualizar()

    # ==========================================================
    # CURAR
    # ==========================================================

    def curar(self):

        herido = min(self.heroes, key=lambda x: x.hp)

        heal = 45
        herido.hp = min(herido.hp_max, herido.hp + heal)

        self.escribir(
            f"💚 Burguesía cura a {herido.nombre} +{heal}"
        )

        self.flash_enemy("green")

        self.turno_enemy()
        self.actualizar()

    # ==========================================================
    # TURNO ENEMIGO
    # ==========================================================

    def turno_enemy(self):

        vivos = [h for h in self.heroes if h.vivo()]

        if vivos and self.enemy.vivo():

            obj = random.choice(vivos)

            dmg = random.randint(20, 38)

            obj.recibir_dano(dmg)

            self.escribir(
                f"👑 El Caudillo ataca a {obj.nombre} ({dmg})"
            )

    # ==========================================================
    # VERIFICAR
    # ==========================================================

    def verificar(self):

        if not self.enemy.vivo():

            messagebox.showinfo(
                "VICTORIA",
                "🎉 ¡Ganaste la batalla!"
            )

        elif all(not h.vivo() for h in self.heroes):

            messagebox.showerror(
                "DERROTA",
                "💀 Tu equipo fue destruido."
            )

    # ==========================================================
    # REINICIAR
    # ==========================================================

    def reiniciar(self):

        self.root.destroy()
        nuevo = tk.Tk()
        Juego(nuevo)
        nuevo.mainloop()


# ==============================================================
# MAIN
# ==============================================================

if __name__ == "__main__":

    root = tk.Tk()
    app = Juego(root)
    root.mainloop()

File: Juego.java
// =============================================================
// RPG SOCIAL MANAGER - VERSION JAVA SWING
// Conversión desde Tkinter Python a Java
// Efectos incluidos:
// ✔ Barras de vida
// ✔ Consola de eventos
// ✔ Ataques
// ✔ Curación
// ✔ Reinicio
// ✔ Interfaz estilo gamer
// =============================================================

import javax.swing.*;
import java.awt.*;
import java.awt.event.*;
import java.util.Random;

public class RPGSocialManager extends JFrame {

    // =========================================================
    // CLASE PERSONAJE
    // =========================================================
    class Personaje {
        String nombre;
        int hp;
        int hpMax;
        int dano;

        public Personaje(String nombre, int hp, int dano) {
            this.nombre = nombre;
            this.hp = hp;
            this.hpMax = hp;
            this.dano = dano;
        }

        public boolean vivo() {
            return hp > 0;
        }

        public void recibirDano(int dmg) {
            hp = Math.max(0, hp - dmg);
        }
    }

    // =========================================================
    // VARIABLES
    // =========================================================

    Personaje est = new Personaje("👨‍🎓 Estudiantes", 160, 24);
    Personaje obr = new Personaje("👷 Obreros", 190, 28);
    Personaje bur = new Personaje("🎩 Burguesía", 145, 20);

    Personaje[] heroes = {est, obr, bur};

    Personaje enemy = new Personaje("👑 El Caudillo", 500, 35);

    JProgressBar[] barras = new JProgressBar[3];
    JLabel[] labelsHP = new JLabel[3];

    JProgressBar barraEnemy;
    JLabel enemyHP;

    JTextArea log;

    Random random = new Random();

    // =========================================================
    // CONSTRUCTOR
    // =========================================================

    public RPGSocialManager() {

        setTitle("⚔ RPG SOCIAL MANAGER ⚔");
        setSize(1250, 760);
        setDefaultCloseOperation(EXIT_ON_CLOSE);
        setLocationRelativeTo(null);
        setLayout(new BorderLayout());
        getContentPane().setBackground(Color.BLACK);

        crearUI();
        actualizar();
    }

    // =========================================================
    // INTERFAZ
    // =========================================================

    public void crearUI() {

        // TITULO
        JLabel titulo = new JLabel("⚔ RPG SOCIAL MANAGER ⚔", SwingConstants.CENTER);
        titulo.setFont(new Font("Arial", Font.BOLD, 28));
        titulo.setForeground(Color.CYAN);
        titulo.setOpaque(true);
        titulo.setBackground(new Color(43, 0, 233));
        add(titulo, BorderLayout.NORTH);

        // PANEL CENTRAL
        JPanel centro = new JPanel(new GridLayout(1, 2));
        centro.setBackground(Color.BLACK);

        // HEROES
        JPanel panelHeroes = new JPanel();
        panelHeroes.setLayout(new GridLayout(3, 1, 10, 10));
        panelHeroes.setBackground(Color.BLACK);

        for (int i = 0; i < heroes.length; i++) {

            JPanel card = new JPanel();
            card.setLayout(new GridLayout(3, 1));
            card.setBackground(new Color(20, 20, 20));
            card.setBorder(BorderFactory.createLineBorder(Color.CYAN, 2));

            JLabel nombre = new JLabel(heroes[i].nombre, SwingConstants.CENTER);
            nombre.setForeground(Color.WHITE);
            nombre.setFont(new Font("Arial", Font.BOLD, 16));

            barras[i] = new JProgressBar(0, heroes[i].hpMax);
            barras[i].setStringPainted(true);

            labelsHP[i] = new JLabel("", SwingConstants.CENTER);
            labelsHP[i].setForeground(Color.GREEN);

            card.add(nombre);
            card.add(barras[i]);
            card.add(labelsHP[i]);

            panelHeroes.add(card);
        }

        // ENEMIGO
        JPanel panelEnemy = new JPanel();
        panelEnemy.setLayout(new GridLayout(3, 1, 10, 10));
        panelEnemy.setBackground(new Color(60, 0, 0));
        panelEnemy.setBorder(BorderFactory.createLineBorder(Color.RED, 3));

        JLabel enemyLabel = new JLabel("👑 EL CAUDILLO 👑", SwingConstants.CENTER);
        enemyLabel.setForeground(Color.RED);
        enemyLabel.setFont(new Font("Arial", Font.BOLD, 22));

        barraEnemy = new JProgressBar(0, enemy.hpMax);
        barraEnemy.setStringPainted(true);

        enemyHP = new JLabel("", SwingConstants.CENTER);
        enemyHP.setForeground(Color.WHITE);

        panelEnemy.add(enemyLabel);
        panelEnemy.add(barraEnemy);
        panelEnemy.add(enemyHP);

        centro.add(panelHeroes);
        centro.add(panelEnemy);

        add(centro, BorderLayout.CENTER);

        // BOTONES
        JPanel botones = new JPanel();
        botones.setBackground(Color.BLACK);

        JButton btnAtaque = crearBoton("⚔ Ataque");
        JButton btnEspecial = crearBoton("✨ Especial");
        JButton btnCurar = crearBoton("💚 Curar");
        JButton btnReset = crearBoton("🔄 Reiniciar");

        botones.add(btnAtaque);
        botones.add(btnEspecial);
        botones.add(btnCurar);
        botones.add(btnReset);

        add(botones, BorderLayout.SOUTH);

        // LOG
        log = new JTextArea(10, 40);
        log.setBackground(Color.BLACK);
        log.setForeground(Color.GREEN);
        log.setFont(new Font("Consolas", Font.PLAIN, 13));
        log.setEditable(false);

        add(new JScrollPane(log), BorderLayout.EAST);

        // EVENTOS
        btnAtaque.addActionListener(e -> atacar());
        btnEspecial.addActionListener(e -> especial());
        btnCurar.addActionListener(e -> curar());
        btnReset.addActionListener(e -> reiniciar());
    }

    // =========================================================
    // BOTON BONITO
    // =========================================================

    public JButton crearBoton(String texto) {

        JButton btn = new JButton(texto);
        btn.setBackground(Color.DARK_GRAY);
        btn.setForeground(Color.WHITE);
        btn.setFocusPainted(false);
        btn.setFont(new Font("Arial", Font.BOLD, 14));

        return btn;
    }

    // =========================================================
    // LOG
    // =========================================================

    public void escribir(String txt) {
        log.append(txt + "\n");
        log.setCaretPosition(log.getDocument().getLength());
    }

    // =========================================================
    // ACTUALIZAR
    // =========================================================

    public void actualizar() {

        for (int i = 0; i < heroes.length; i++) {
            barras[i].setValue(heroes[i].hp);
            labelsHP[i].setText(heroes[i].hp + "/" + heroes[i].hpMax);
        }

        barraEnemy.setValue(enemy.hp);
        enemyHP.setText(enemy.hp + "/" + enemy.hpMax);
    }

    // =========================================================
    // ATAQUE
    // =========================================================

    public void atacar() {

        for (Personaje h : heroes) {
            if (h.vivo()) {

                int dmg = random.nextInt(11) + (h.dano - 5);

                enemy.recibirDano(dmg);

                escribir(h.nombre + " golpea por " + dmg);
            }
        }

        turnoEnemy();
        verificar();
        actualizar();
    }

    // =========================================================
    // ESPECIAL
    // =========================================================

    public void especial() {

        int total = random.nextInt(41) + 90;

        enemy.recibirDano(total);

        escribir("✨ ATAQUE COMBINADO causa " + total + " daño");

        turnoEnemy();
        verificar();
        actualizar();
    }

    // =========================================================
    // CURAR
    // =========================================================

    public void curar() {

        Personaje herido = heroes[0];

        for (Personaje h : heroes) {
            if (h.hp < herido.hp) {
                herido = h;
            }
        }

        int heal = 45;
        herido.hp = Math.min(herido.hpMax, herido.hp + heal);

        escribir("💚 Burguesía cura a " + herido.nombre + " +" + heal);

        turnoEnemy();
        actualizar();
    }

    // =========================================================
    // TURNO ENEMIGO
    // =========================================================

    public void turnoEnemy() {

        if (!enemy.vivo()) return;

        Personaje objetivo = null;

        while (objetivo == null) {
            Personaje temp = heroes[random.nextInt(heroes.length)];
            if (temp.vivo()) objetivo = temp;

            boolean todosMuertos = true;
            for (Personaje h : heroes)
                if (h.vivo()) todosMuertos = false;

            if (todosMuertos) return;
        }

        int dmg = random.nextInt(19) + 20;

        objetivo.recibirDano(dmg);

        escribir("👑 El Caudillo ataca a " + objetivo.nombre + " (" + dmg + ")");
    }

    // =========================================================
    // VERIFICAR
    // =========================================================

    public void verificar() {

        if (!enemy.vivo()) {
            JOptionPane.showMessageDialog(this,
                    "🎉 ¡Ganaste la batalla!");
        }

        boolean muertos = true;

        for (Personaje h : heroes)
            if (h.vivo()) muertos = false;

        if (muertos) {
            JOptionPane.showMessageDialog(this,
                    "💀 Tu equipo fue destruido.");
        }
    }

    // =========================================================
    // REINICIAR
    // =========================================================

    public void reiniciar() {
        dispose();
        new RPGSocialManager().setVisible(true);
    }

    // =========================================================
    // MAIN
    // =========================================================

    public static void main(String[] args) {

        SwingUtilities.invokeLater(() -> {
            new RPGSocialManager().setVisible(true);
        });
    }
}
